### DISCUSSION TITLE
How to configure Wastebin?

### DISCUSSION THREAD

**Discussion by Julian0o** [tteck/Proxmox#3148] [Help Needed]
Hi there,

I installed Wastebin and like to configure the variables like BASE_URL, BRAND_NAME etc. 
I didn't found a config file in the lxc. How can i do that?

<img width="846" alt="image" src="https://github.com/tteck/Proxmox/assets/7078907/a9e9c403-5383-4cf9-b1f2-59a5d1360a54">

**Comment by tteck** [comment]
Try adding the Environment Variables to the `wastebin.service` file. 

e.g.
```
 [Unit]
Description=Start Wastebin Service
After=network.target

[Service]
User=root
Environment=BRAND_NAME=Helper-Scripts
WorkingDirectory=/opt/wastebin
ExecStart=/root/.cargo/bin/cargo run --release --quiet

[Install]
WantedBy=multi-user.target
```

**Reply by Julian0o** [reply]
I edited the file: /etc/systemd/system/wastebin.service
```
[Unit]
Description=Start Wastebin Service
After=network.target

[Service]
User=root
Environment=PAGE_TITLE=PasteBin
WorkingDirectory=/opt/wastebin
ExecStart=/root/.cargo/bin/cargo run --release --quiet

[Install]
WantedBy=multi-user.target`

After hat: 
systemctl daemon-reload
systemctl restart wastebin.service
```

But it did not work.

**Reply by tteck** [reply]
After editing the file, did you

```
systemctl daemon-reload
systemctl restart wastebin
```

**Reply by tteck** [reply]
I now see that you did.

**Reply by kamaradski** [reply]
this worked fine for me, thanks!

`[Unit]
Description=Start Wastebin Service
After=network.target

[Service]
User=root
Environment="WASTEBIN_TITLE=RinBinBin"
Environment="WASTEBIN_HTTP_TIMEOUT=60"
Environment="WASTEBIN_MAX_BODY_SIZE=100"
WorkingDirectory=/opt/wastebin
ExecStart=/root/.cargo/bin/cargo run --release --quiet

[Install]
WantedBy=multi-user.target`

**Comment by Mellowlynx** [comment]
Be sure to punt in the correct environment name, to update the title, use `WASTEBIN_TITLE`
Also put in like this
```
Environment="WASTEBIN_TITLE=PasteBin"
```

I made a service override file, to be extra safe and keep changes after updates.

Run `systemctl edit wastebin`, which will create an override file for you or let you edit an existing one.
It will be saved as `/etc/systemd/system/wastebin.service.d/override.conf`

Only add in there:
```
[Service]
Environment="WASTEBIN_TITLE=PasteBin"
```

Save, reload demon, and reload wastebin service.
This should do it.
